src/haz3lcore/dynamics/DH.re

```diff
@@ -1,52 +1,55 @@
 open Sexplib.Std;
+open Hazel_menhir.AST;
 
 [@deriving (show({with_path: false}), sexp, yojson)]
 type if_consistency =
   | ConsistentIf
   | InconsistentIf;
 
 module rec DHExp: {
+  //All comments show the textual syntax for DHExp when built using the menhir parser
   [@deriving (show({with_path: false}), sexp, yojson)]
   type t =
-    | EmptyHole(MetaVar.t, HoleInstanceId.t)
-    | NonEmptyHole(ErrStatus.HoleReason.t, MetaVar.t, HoleInstanceId.t, t)
-    | FreeVar(MetaVar.t, HoleInstanceId.t, Var.t)
+    | EmptyHole(MetaVar.t, HoleInstanceId.t) //()
+    | NonEmptyHole(ErrStatus.HoleReason.t, MetaVar.t, HoleInstanceId.t, t) //(_HOLE x)
+    | FreeVar(MetaVar.t, HoleInstanceId.t, Var.t) //(_FREE x)
     | InvalidText(MetaVar.t, HoleInstanceId.t, string)
-    | InconsistentBranches(MetaVar.t, HoleInstanceId.t, case)
+    | InconsistentBranches(MetaVar.t, HoleInstanceId.t, case) //(_HOLE (case ...))
```

Comment:
let's use `{{?case ...}}` for this one

Reply:
Sorry, my docs were completely wrong for the syntax of that one. It was just ( case ... ) but I'll change it to the {{?case ...}}